core/templates/pages/topic-editor-page/subtopic-editor/subtopic-preview-tab.component.spec.ts

```diff
@@ -0,0 +1,203 @@
+    expect(component.subtopicPage).toEqual(subtopicPage);
+    expect(component.pageContents).toEqual(subtopicPageContentsDict);
+    expect(component.htmlData).toEqual('test content');
+  });
+
+  it('should get subtopic contents when subtopic preview page is' +
+  ' loaded', () => {
+    spyOnProperty(topicEditorStateService, 'onSubtopicPageLoaded').and
+      .returnValue(subtopicPageLoadedEventEmitter);
+    // The ngOnInit is called to add the directiveSubscriptions.
+    component.ngOnInit();
+    component.subtopicPage = null;
+    component.pageContents = null;
+    component.htmlData = '';
+
+    subtopicPageLoadedEventEmitter.emit();
+
+    expect(component.subtopicPage).toEqual(subtopicPage);
+    expect(component.pageContents).toEqual(subtopicPageContentsDict);
+    expect(component.htmlData).toEqual('test content');
+  });
+
+  it('should call initEditor when topic is initialized', () => {
+    spyOnProperty(topicEditorStateService, 'onTopicInitialized').and
+      .returnValue(topicInitializedEventEmitter);
+    // The ngOnInit is called to add the directiveSubscriptions.
+    component.ngOnInit();
+    component.subtopicId = 2;
+    component.editableTitle = 'random title';
+    component.editableThumbnailFilename = 'random_file_name.svg';
+
+    topicInitializedEventEmitter.emit();
+
+    expect(component.subtopicId).toBe(1);
+    expect(component.editableTitle).toBe('Subtopic1');
+    expect(component.editableThumbnailFilename).toBe('thumbnailFilename.svg');
+  });
+
+  it('should call initEditor when topic is reinitialized', () => {
+    spyOnProperty(topicEditorStateService, 'onTopicInitialized').and
+      .returnValue(topicInitializedEventEmitter);
+    spyOnProperty(topicEditorStateService, 'onTopicReinitialized').and
+      .returnValue(topicReinitializedEventEmitter);
+    // The ngOnInit is called to add the directiveSubscriptions.
+    component.ngOnInit();
+
+    topicInitializedEventEmitter.emit();
+    // Change values.
+    component.subtopicId = 2;
+    component.editableTitle = 'random title';
+    component.editableThumbnailFilename = 'random_file_name.svg';
+    topicReinitializedEventEmitter.emit();
+
+    expect(component.subtopicId).toBe(1);
+    expect(component.editableTitle).toBe('Subtopic1');
+    expect(component.editableThumbnailFilename).toBe('thumbnailFilename.svg');
+  });
+
+  it('should navigate to subtopic editor when subtopic is clicked', () => {
+    spyOn(topicEditorRoutingService, 'navigateToSubtopicEditorWithId');
+
+    // Pre-check.
+    expect(component.subtopicId).not.toBeDefined();
+
+    component.ngOnInit();
+    component.navigateToSubtopic();
+
+    expect(component.subtopicId).toBe(1);
+    expect(topicEditorRoutingService.navigateToSubtopicEditorWithId)
+      .toHaveBeenCalledWith(1);
+  });
+
+  it('should set thumbnailIsShown to true if thumbnail is displayed', () => {
+    component.changeContent('thumbnail');
+
+    expect(component.thumbnailIsShown).toBe(true);
+  });
```

Comment:
Can we test the interface rather than implementation details?
(Ex: `should display thumbnail when x is triggered`)
Missing pre-checks